docs/architecture/adr/0015-short-lived-browser-services.md

```diff
@@ -0,0 +1,154 @@
+---
+adr: "0015"
+status: In progress
+date: 2022-12-09
+tags: [browser, clients, typescript]
+---
+
+# 0015 - Short Lived Browser Services
+
+<AdrTable frontMatter={frontMatter}></AdrTable>
+
+## Context and Problem Statement
+
+[ADR 003](./0003-observable-data-services.md) introduced the usage of Observables into our
+TypeScript codebase. That same ADR describes using `ngDestroy` to trigger unsubscribing from all
+subscribed `Observables`. However, `ngDestroy` is only called when the Angular router is used to
+navigate away from the page. In SPAs, this isn't a problem. If the router isn't used, it means you
+closed the page and the SPA is dead. In Browser Extensions, we have a persistent background that
+survives such closing events. This means that subscriptions exist in the Observers queue which no
+longer exist.
+
+In FireFox, in particular, this is a catastrophic failure. FireFox will swap out all object
+belonging to a DOM object that no longer exists with a `DeadObject`. When `next()` is called on the
+`Subject`, the `DeadObject` is treated like and Observer and throws an error, preventing any further
+notification to subsequent Observers and catastrophic breaks in the extension.
```

Comment:
This seems a bit dramatic?

```suggestion
notification to subsequent Observers and causes catastrophic breaks in the extension.
```

Reply:
I don't think so, I chose that word due to the need to fully reboot the browser to fix. We could say irrecoverable, but they are roughly synonyms to me in this context 🤷